demo/src/components/heatmap/DataGrid.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const DataGrid = ({
+  xLabels,
+  yLabels,
+  data,
+  xLabelWidth,
+  boxSize,
+  background,
+  height,
+}) => {
+  const flatArray = data.reduce((i, o) => [...o, ...i], []);
+  const max = Math.max(...flatArray);
+  const min = Math.min(...flatArray);
+  return (
+    <div style={{"white-space": "nowrap"}}>
+      {yLabels.map((y, yi) => (
+        <div key={`${y}_${yi}`} style={{clear: "both"}}>
+          <div style={{display: "inline-block", width: xLabelWidth, textAlign: 'right', paddingRight: '5px', paddingTop:`${boxSize/3.7}px`}}>{y}</div>
+          {xLabels.map((x, xi) => (
+            <div
+              title={`${data[yi][xi]}`}
+              key={`${x}_${xi}_${y}_${yi}`}
+              style={{
```

Comment:
Is it possible to use a class rather than inline style?  This applies a few lines above as well.

Reply:
I have no idea.  I don't know enough about how react actually works.  It'd be nice to have a front-end developer...

Reply:
You could add a class="foo" tag and instead make this a css entry foo in main.css or another included css file. This makes the style easier to understand and configure. A frontend dev would cringe at these inline style tags, but what you have is fine for now.

Reply:
For some of these, that would not work, because we want to be able to configure sizes programmatically.  I have different heatmaps in the semantic parser demo with different x label widths, for instance.

I also was a little annoyed that we're just using plain CSS instead of something like less that allows variables, so I didn't have much motivation to put this into a flat CSS file.  My vote is to get a front-end developer who can clean all of this up.